From: Chuck Lever <chuck.lever@oracle.com>
Date: Tue, 27 Mar 2018 10:51:22 -0400
Subject: [PATCH] sunrpc: Re-purpose trace_svc_process
Git-commit: 0b9547bf6b94317b3f8e2496dc2b44cb6e599b01
Patch-mainline: v4.17
References: bsc#1205006

Currently, trace_svc_process has two call sites:

1. Just after a call to svc_send. svc_send already invokes
   trace_svc_send with the same arguments just before returning

2. Just before a call to svc_drop. svc_drop already invokes
   trace_svc_drop with the same arguments just after it is called

Therefore trace_svc_process does not provide any additional
information not already provided by these other trace points.

However, it would be useful to record the incoming RPC procedure.
So reuse trace_svc_process for this purpose.

[[ NOTE this version of the patch for SLE-LTSS kernels just
   removes the current trace_svc_process call because it isn't
   helpful and actually causes a crash.
   The new trace isn't added as it isn't that imporant and we
   keep changes as small as possible for LTSS
   - NeilBrown
]]

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 include/trace/events/sunrpc.h |    4 ----
 net/sunrpc/svc.c              |    8 ++------
 2 files changed, 2 insertions(+), 10 deletions(-)

--- a/include/trace/events/sunrpc.h
+++ b/include/trace/events/sunrpc.h
@@ -535,10 +535,6 @@ DECLARE_EVENT_CLASS(svc_rqst_status,
 		__entry->status, show_rqstp_flags(__entry->flags))
 );
 
-DEFINE_EVENT(svc_rqst_status, svc_process,
-	TP_PROTO(struct svc_rqst *rqst, int status),
-	TP_ARGS(rqst, status));
-
 DEFINE_EVENT(svc_rqst_status, svc_send,
 	TP_PROTO(struct svc_rqst *rqst, int status),
 	TP_ARGS(rqst, status));
--- a/net/sunrpc/svc.c
+++ b/net/sunrpc/svc.c
@@ -1452,14 +1452,10 @@ svc_process(struct svc_rqst *rqstp)
 	}
 
 	/* Returns 1 for send, 0 for drop */
-	if (likely(svc_process_common(rqstp, argv, resv))) {
-		int ret = svc_send(rqstp);
+	if (likely(svc_process_common(rqstp, argv, resv)))
+		return svc_send(rqstp);
 
-		trace_svc_process(rqstp, ret);
-		return ret;
-	}
 out_drop:
-	trace_svc_process(rqstp, 0);
 	svc_drop(rqstp);
 	return 0;
 }
